packages/mgt-components/src/components/mgt-file-list/mgt-file-list.ts

```diff
@@ -0,0 +1,592 @@
+/**
+ * -------------------------------------------------------------------------------------------
+ * Copyright (c) Microsoft Corporation.  All Rights Reserved.  Licensed under the MIT License.
+ * See License in the project root for license information.
+ * -------------------------------------------------------------------------------------------
+ */
+
+import { arraysAreEqual, MgtTemplatedComponent, Providers, ProviderState } from '@microsoft/mgt-element';
+import * as MicrosoftGraph from '@microsoft/microsoft-graph-types';
+import { customElement, html, property, TemplateResult } from 'lit-element';
+import { debounce } from '../../utils/Utils';
+import { repeat } from 'lit-html/directives/repeat';
+import {
+  getDriveFilesById,
+  getDriveFilesByPath,
+  getFiles,
+  getFilesById,
+  getFilesByListQuery,
+  getFilesByPath,
+  getFilesByQueries,
+  getGroupFilesById,
+  getGroupFilesByPath,
+  getMyInsightsFiles,
+  getSiteFilesById,
+  getSiteFilesByPath,
+  getUserFilesById,
+  getUserFilesByPath,
+  getUserInsightsFiles
+} from '../../graph/graph.files';
+
+import { OfficeGraphInsightString, ViewType } from '../../graph/types';
```

Comment:
I should have asked this in the file PR, what do you think about renaming this to `InsightType` instead of `OfficeGraphInsightString`?

Reply:
The interface for insights are as this. I wanted to follow the pattern from graph entity. Let me know what you think
```
export interface OfficeGraphInsights extends Entity {
    // Access this property from the derived type itemInsights.
    shared?: NullableOption<SharedInsight[]>;
    // Access this property from the derived type itemInsights.
    trending?: NullableOption<Trending[]>;
    // Access this property from the derived type itemInsights.
    used?: NullableOption<UsedInsight[]>;
}
```